Week-6/Assignment-5/index.js

```diff
@@ -0,0 +1,52 @@
+const A1 = [5, 10, 3, 2, 50, 80], B1 = 78;
+const A2 = [-10, 20], B2 = 30;
+const A3 = [8, 0, 2, 0, 3, 9, 10, 50, 103], B3 = 103;
+const A4 = [8, 0, 2, 0, 3, 9, 10, 50, 103], B4 = -106;
+const A5 = [-104, 20], B5 = 20;
+
+/**
+ * @function validateInput checks the input for 1 <= N <= 105 -103 <= A[i] <= 103 -105 <= B <= 105
+ * @param {Array} arr an array
+ * @returns boolean
+ */
+const validateInput = (arr, diff) => Array.isArray(arr) && arr.length > 1 && arr.length < 105 &&
+    arr.every(el => Number.isInteger(el) && el >= -103 && el <= 103) &&
+    Number.isInteger(diff) && diff >= -105 && diff <= 105;
+
+/**
+ * @function pairWithDiff finds a pair with a given difference if available
+ * @summary Time Complexity - O(n), Space Complexity - O(1)
+ * @param {Array} arr an input array
+ * @param {Number} diff a difference to find a pair with
+ * @returns 1 or 0 based on if a pair with given difference is found
+ */
+const pairWithDiff = (arr, diff) => {
+    if (!validateInput(arr, diff)) throw new Error('Invalid input provided');
+
+    arr = arr.sort((a, b) => a - b);
+    const arrLen = arr.length;
+
+    for (let i = 0; i < arrLen; i++) {
+        for (let j = i; j < arrLen; j++) {
```

Comment:
@tusharwalzade216 Shouldn't j start from i + 1 ?
Also, Since there are two loops running, how is the time complexity O(n) ?